packages/gatsby-telemetry/src/telemetry.js

```diff
@@ -0,0 +1,196 @@
+      ...tags, // The schema must include these
+      eventType,
+      sessionId: this.sessionId,
+      time: new Date(),
+      machineId: this.getMachineId(),
+      repositoryId: this.getRepoId(),
+      componentId: `gatsby-cli`,
+      osInformation: this.getOsInfo(),
+      componentVersion: this.componentVersion,
+    }
+    this.store.addEvent(event)
+  }
+
+  getMachineId() {
+    // Cache the result
+    if (this.machineId) {
+      return this.machineId
+    }
+    let machineId = this.store.getConfig(`machineId`)
+    if (!machineId) {
+      machineId = uuid()
+      this.store.updateConfig({ machineId })
+    }
+    this.machineId = machineId
+    return machineId
+  }
+
+  isTrackingEnabled() {
+    // Cache the result
+    if (this.trackingEnabled !== undefined) {
+      return this.trackingEnabled
+    }
+    let telemetry = this.store.getConfig(`telemetry`)
+    if (telemetry === undefined || telemetry === null) {
+      console.log(
+        `Gatsby has started collecting anonymous usage analytics to help improve Gatsby for all users.\n` +
+          `If you'd like to opt-out, you can use \`gatsby analytics --disable\`\n` +
+          `To learn more, checkout http://gatsby.dev/analytics`
+      )
+      telemetry = true
+      this.store.updateConfig({ telemetry })
+    }
+    this.trackingEnabled = telemetry
+    return telemetry
+  }
+
+  getRepoId() {
+    // we may live multiple levels in git repo
+    let prefix = `pwd:`
+    let repo = basename(process.cwd())
+    try {
+      const originBuffer = execSync(
+        `git config --local --get remote.origin.url`,
+        { timeout: 1000 }
+      )
+      repo = String(originBuffer).trim()
+      prefix = `git:`
+    } catch (e) {
+      // ignore
+    }
+    const hash = createHash(`sha256`)
+    hash.update(repo)
+    return prefix + hash.digest(`hex`)
+  }
+
+  getOsInfo() {
+    if (this.osInfo) {
+      return this.osInfo
+    }
+    const cpus = os.cpus()
+    const osInfo = {
+      nodeVersion: process.version,
+      platform: os.platform(),
+      release: os.release(),
+      cpus: cpus && cpus.length > 0 && cpus[0].model,
+      arch: os.arch(),
+      ci: ci.isCI,
+      ciName: (ci.isCI && ci.name) || undefined,
+    }
+    this.osInfo = osInfo
+    return osInfo
+  }
+
+  trackActivity(source) {
+    if (!this.isTrackingEnabled()) {
+      return
+    }
+    // debounce by sending only the first event whithin a rolling window
+    const now = Date.now()
+    const last = this.debouncer[source] || 0
+    const debounceTime = 5 * 1000 // 5 sec
+
+    if (now - last > debounceTime) {
+      this.captureEvent(source)
+    }
+    this.debouncer[source] = now
+  }
+
+  decorateNextEvent(event, obj) {
+    const cached = this.metadataCache[event] || {}
+    this.metadataCache[event] = Object.assign(cached, obj)
+  }
+
+  decorateAll(tags) {
+    this.defaultTags = Object.assign(this.defaultTags, tags)
+  }
+
+  setAnalyticsEnabled(enabled) {
```

Comment:
```suggestion
  setTelemetryEnabled(enabled) {
```

Seems like it was missed in analytics -> telemetry change (this is called with `setTelemtryEnabled` in https://github.com/gatsbyjs/gatsby/pull/12758/files#diff-54b3fd6936bf2f005bf1de5eb5ea79e0R24

Reply:
Thanks!